runtime/parachains/src/paras_inherent.rs

```diff
@@ -743,13 +767,24 @@ fn apply_weight_limit<T: Config + inclusion::Config>(
 		return total
 	}
 
+	// Prefer code upgrades, they tend to be large and hence stand no chance to be picked
+	// late while maintaining the weight bounds
+	let preferred_indices = candidates
```

Comment:
Yep, but this also means that by means of this a single malicious Parachain (e.g. hacked collator), could DOS all other parachains. A solution would be some kind of rate limiting for parachain runtime upgrades, longer term we really want them to be off-chain. The question is how quickly can we get that done.

Reply:
Created a tracking issue https://github.com/paritytech/polkadot/issues/4335